omegaconf/dictconfig.py

```diff
@@ -186,6 +188,8 @@ def is_typed(c: Any) -> bool:
         if is_dict(value_type) and is_dict(target_type):
             return
 
+        if is_generic_list(target_type) or is_generic_dict(target_type):
+            return
```

Comment:
Can I assign a List[str] to a Dict[str, str] now?

Reply:
No you can't.